svir/dialogs/viewer_dock.py

```diff
@@ -373,6 +373,24 @@ def filter_dmg_by_asset_aggr(self):
             return
         self.draw_dmg_by_asset_aggr()
 
+    def filter_agg_curves(self):
+        params = {}
+        for tag_name in self.tags:
+            if self.tags[tag_name]['selected']:
+                for value in self.tags[tag_name]['values']:
+                    if self.tags[tag_name]['values'][value]:
+                        # NOTE: this would not work for multiple values per tag
+                        params[tag_name] = value
+        to_extract = 'agg_curves/%s' % self.loss_type_cbx.currentText()
+        with WaitCursorManager(
+                'Extracting...', message_bar=self.iface.messageBar()):
+            self.agg_curves = extract_npz(
+                self.session, self.hostname, self.calc_id, to_extract,
+                message_bar=self.iface.messageBar(), params=params)
+        if self.agg_curves is None:
```

Comment:
In which case this happens?

Reply:
I think you are right. This should never happen, or it should break in case it happens.

Reply:
I've added a check while drawing, that clears the plot in case too much filtering produces an empty array.